src/Controllers/API/UpdateStudentProfileController.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace Portal\Controllers\API;
+
+use Exception;
+use Portal\Abstracts\Controller;
+use Portal\Interfaces\ApplicantModelInterface;
+use Portal\Models\ApplicantModel;
+use Portal\Sanitisers\ApplicantSanitiser;
+use Portal\Validators\ApplicantValidator;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class UpdateStudentProfileController extends Controller
+{
+    private ApplicantModel $applicantModel;
+
+    public function __construct(ApplicantModel $applicantModel)
+    {
+        $this->applicantModel = $applicantModel;
+    }
+
+    public function __invoke(Request $request, Response $response, array $args): Response
+    {
+        $updatedStudentProfileData = $request->getParsedBody();
+
+        if (
+            !empty($_SESSION['studentLogin']) &&
+            $_SESSION['studentLogin'] &&
+            $_SESSION['studentId'] == $updatedStudentProfileData['id'] ||
+            !empty($_SESSION['loggedIn']) &&
+            $_SESSION['loggedIn']
+        ) {
+            $responseBody = [];
+
+            if (array_key_exists("edaid", $updatedStudentProfileData)) {
```

Comment:
i would like the remainder of the method __invoke to look something like this:
```php
    if (array_key_exists("edaid", $updatedStudentProfileData)) {
        $this->updateEdaid($updatedStudentProfileData);
    } elseif (array_key_exists("upfront", $updatedStudentProfileData)) {
        $this->updateUpfront($updatedStudentProfileData);
    } elseif (array_key_exists("githubUsername", $updatedStudentProfileData)) {
        $this->updateGithubUsername($updatedStudentProfileData);
    } elseif (array_key_exists("laptop", $updatedStudentProfileData)) {
         $this->updateLaptop($updatedStudentProfileData);
    } else {
        $responseBody["success"] = false;
        $responseBody["msg"] = "Unsupported field";
        $responseBody["data"] = [];
        $statusCode = 400;
        return $this->respondWithJson($response, $responseBody, $statusCode);
    }
}
return $response->withStatus(401);
```

basically i want you to break up your huge method into multiple smaller methods so the code is more readable